### PR TITLE
Allow rule go_library to accept ASM source files

### PR DESCRIPTION
**Pull request by linuxerwang**
…in Go; add examples for go_asm_library; change go_library to take multiple cgo_objects; update README for go_asm_library and go_library.

This replaces the pull request #22 with a new implementation since I messed it up. The new impl followed Yugui's suggestions.

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

File: examples/lib/lib_test.go [hunk 1]
```diff
@@ -24,3 +24,7 @@ func TestMeaning(t *testing.T) {
 		t.Errorf("got %d, want 42", m)
 	}
 }
+
+// func TestAddTwoNumbers(t *testing.T) {
```
Comment (linuxerwang) [inline examples/lib/lib_test.go]: I commented out this test because it fails. I need bazel team's help to make it work.

**Comment by damienmg** [comment]
Thank for the contribution. This change looks pretty good but I would like to get the opinion of someone more familiar with the go toolchain @yugui, can you have a look at it?

Also I think you need to fix the author line on your commit so that @googlebot is happy.

Finally, for Jenkins:
test this please

File: examples/lib/lib_test.go (hunk 1 above)
Comment (damienmg) [inline examples/lib/lib_test.go]: Which failure do you get?

Comment (linuxerwang) [inline examples/lib/lib_test.go]: github.com/bazelbuild/rules_go/examples/lib/lib_test.AddTwoNumbers: call to external function github.com/bazelbuild/rules_go/examples/lib/lib_test.add
github.com/bazelbuild/rules_go/examples/lib/lib_test.AddTwoNumbers: github.com/bazelbuild/rules_go/examples/lib/lib_test.add: not defined
github.com/bazelbuild/rules_go/examples/lib/lib_test.AddTwoNumbers: undefined: github.com/bazelbuild/rules_go/examples/lib/lib_test.add

**Comment by linuxerwang** [comment]
Thank you for the help, damienmg. The author line has been fixed.

**Comment by yugui** [comment]
@linuxerwang 
You needed to propagate asm sources to "lib_test" to build the target.

I've done three things in linuxerwang/rules_go#1:
* Propagate the source via `go_library.library` attr
* Reduce function call depth of `emit_go_asm_action`
* Simplify `emit_go_asm_action`

**Comment by linuxerwang** [comment]
@yugui Thank you so much.

**Comment by yugui** [comment]
* The commit message of your first commit no longer reflects the content of the change
* It'd better to squash my commits into your one

**Comment by linuxerwang** [comment]
Done. Thanks!

**Comment by damienmg** [comment]
Jenkins, test this please.

**Comment by damienmg** [comment]
LGTM if yugui@ is good I can squash and merge.

**Comment by linuxerwang** [comment]
@yugui could you please sign off for a merge? Thank you.

**Comment by yugui** [comment]
Yes.  It looks good to me. @damienmg, could you merge it?

**Pull request merged**